engine/src/lib.rs

```diff
@@ -143,6 +143,20 @@ mod contract {
         let state = engine::get_state(&io).sdk_unwrap();
         io.return_output(state.owner_id.as_bytes());
     }
+    
+    /// Set owner account id for this contract.
+    #[no_mangle]
+    pub extern "C" fn set_owner() {
+        let mut io = Runtime;
+        let mut state = engine::get_state(&io).sdk_unwrap();
+        require_owner_only(&state, &io.predecessor_account_id());
+        let args = io.read_input().to_vec();
+        let sr: String = String::from_utf8(args).unwrap();
```

Comment:
There is no need to create a string here. You can create `AccoundId` from `Vec<u8>` here:
```rust
let args = io.read_input().to_vec();
let id = AccountId::try_from(args).unwrap();
```        
`AccountId` has implementations `TryFrom<Vec<u8>>` and `TryFromo<&[u8]>` traits.